src/client/opamArg.ml

```diff
@@ -782,28 +774,263 @@ let opamlist_columns =
   in
   parse, print
 
-(* Helpers *)
-let mk_flag ?section flags doc =
+(* CLI version helpers *)
+
+type flag_validity =
+  | Valid_since of OpamCLIVersion.t
+  | Deprecated of OpamCLIVersion.t * OpamCLIVersion.t * string option
+  (* since * removal * use instead *)
+
+let cli2_0 = OpamCLIVersion.of_string "2.0"
+let cli2_1 = OpamCLIVersion.of_string "2.1"
+let cli2_3 = OpamCLIVersion.of_string "2.3"
+let cli3_0 = OpamCLIVersion.of_string "3.0"
+let valid_cli_legacy = Valid_since cli2_0
```

Comment:
These are dropped here, but maybe there is a better place to put them. I thought about directly in `OpamCLIVersion`, but it would be quite more verbose to write/read.